### PR TITLE
set token cache to 30 minutes

### PR DESCRIPTION
**Pull request by cindy-broadinstitute**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/4948

## Summary of changes

Adjust the user token caching logic to be less aggressive, so that requests made with an almost expire token will trigger a token refresh instead of using the cached one

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

**Review by cpate4 — APPROVED** [review]
:shipit:

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/CloudServiceAuthTokenProviderSpec.scala [hunk 1]
```diff
@@ -27,7 +27,8 @@ class CloudServiceAuthTokenProviderSpec
     val result = for {
       now <- IO.realTimeInstant
     } yield {
-      tokens.push(CloudToken(secondTokenValue, now.plusSeconds(10)))
+      // set secondTokenValue to greater than the tokenCacheTimeInMilli which is 30 minutes
+      tokens.push(CloudToken(secondTokenValue, now.plusSeconds(40 * 60)))
```
Comment (LizBaldo) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/CloudServiceAuthTokenProviderSpec.scala]: 30 * 60 + 10?

**Review by LizBaldo — APPROVED** [review]
lgtm, I just have one suggestion about the unit test

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/CloudServiceAuthTokenProviderSpec.scala (hunk 1 above)
Comment (cindy-broadinstitute) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/CloudServiceAuthTokenProviderSpec.scala]: Sure, that might be clearer. Though 40 minutes covers it ;)

**Review by giventocode — APPROVED** [review]
LGTM.

**Pull request merged**